From 7ee068b3da7e8b60a6c0a5c95b9026ef2c82ebb7 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Fri, 5 Jul 2024 15:03:09 +0200
Subject: [PATCH] Fix SONNET_USE_QML=OFF by moving ECMQmlModule behind the
 conditional

Otherwise we still end up with

> -- Could NOT find Qt6Qml (missing: Qt6Qml_DIR)

and finally

> Failed setting up ECMQmlModule

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3427bb12..892f77d7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,6 @@ include(ECMSetupVersion)
 include(ECMGenerateHeaders)
 include(ECMGenerateQmlTypes)
 include(ECMFindQmlModule)
-include(ECMQmlModule)
 include(ECMDeprecationSettings)
 include(ECMAddQch)
 include(ECMPoQmTools)
@@ -39,6 +38,7 @@ if(SONNET_USE_WIDGETS)
 endif()
 
 if(SONNET_USE_QML)
+    include(ECMQmlModule)
     find_package(Qt6 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Quick)
 endif()
 
-- 
2.45.2

